spec/ameba/presenter/rule_versions_presenter_spec.cr

```diff
@@ -0,0 +1,34 @@
+require "../../spec_helper"
+
+module Ameba
+  private def with_rule_versions_presenter(&)
+    with_presenter(Presenter::RuleVersionsPresenter) do |presenter, io|
+      rules = Config.load.rules
+      presenter.run(rules)
+
+      output = io.to_s
+      output = Formatter::Util.deansify(output).to_s
+
+      yield rules, output, presenter
+    end
+  end
+
+  describe Presenter::RuleVersionsPresenter do
+    it "outputs rule versions" do
+      with_rule_versions_presenter do |rules, output|
+        missing_version = SemanticVersion.new(0, 0, 0)
+        versions = rules
+          .sort_by { |rule| rule.since_version || missing_version }
+          .group_by(&.since_version)
+
+        versions.each do |version, version_rules|
+          output.should contain (version || "N/A").to_s
+
+          version_rules.map(&.name).each do |name|
+            output.should contain name
+          end
```

Comment:
**thought:** This is close to a re-implementation of `RuleVersionsPresenter`. If both have the same logic error, the testing is pretty useless. Also the tests do not establish any relation between the individual items, so there could be many wrong outputs that still pass the test.
Why not test against a fixed output string?